src/main/java/org/opentripplanner/middleware/otp/response/Place.java

```diff
@@ -31,18 +31,32 @@
     public Set<String> networks;
     public String address;
 
+    /**
+     * This method calculates equality in the context of trip monitoring in order to analyzing equality when
+     * checking if itineraries are the same.
+     */
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         Place place = (Place) o;
+        // FIXME account for slight stop repositioning by calculating equality based off of proximity to previous stop
         return lon.equals(place.lon) &&
             lat.equals(place.lat) &&
-            Objects.equals(stopId, place.stopId);
+            Objects.equals(name, place.name);
```

Comment:
My concern with the stopId is that it could change due to an agency changing their internal IDs. I'll change it back for now, but also add a comparison with vertexType as that will differentiate between non-transit vertices.